kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/embedder/MavenEmbedder.java

```diff
@@ -292,18 +300,29 @@ public String getLocation() {
 
         ClassLoader originalCl = Thread.currentThread().getContextClassLoader();
         try {
-            Thread.currentThread().setContextClassLoader( componentProvider.getSystemClassLoader() );
-            ProjectBuilder projectBuilder = componentProvider.lookup( ProjectBuilder.class );
-            // BZ-1007894: Check if added dependencies are resolvable.
-            ProjectBuildingResult result = projectBuilder.build( modelSource, getProjectBuildingRequest() );
-            if ( result != null && result.getDependencyResolutionResult() != null && !result.getDependencyResolutionResult().getCollectionErrors().isEmpty() ) {
-                // A dependency resolution error has been produced. It can contains some error. Throw the first one to the client, so the user will fix every one sequentially.
-                Exception depedencyResolutionException = result.getDependencyResolutionResult().getCollectionErrors().get( 0 );
-                if ( depedencyResolutionException != null ) {
-                    throw new MavenEmbedderException( depedencyResolutionException.getMessage(), depedencyResolutionException );
+            org.eclipse.aether.artifact.Artifact lastArtifact = null;
+            do {
```

Comment:
I tested this locally and the scenario described in AF-1646 is working fine with this fix.
I kinda don't like how convoluted this code is after the changes. infinite do-while loop? continue? mutable variables to control state of the looping? No comments explaining why we tryRemoveLocalArtifact?

I think this code could be made more readable

Reply:
@jhrcek could you provide a suggestion? as for me, the code looks ok (of course I'm biased here).